src/main/java/com/github/mizool/jediscache/JedisCacheManager.java

```diff
@@ -0,0 +1,231 @@
+package com.github.mizool.jediscache;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import javax.cache.Cache;
+import javax.cache.CacheException;
+import javax.cache.CacheManager;
+import javax.cache.configuration.CompleteConfiguration;
+import javax.cache.configuration.Configuration;
+
+import lombok.Data;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Lists;
+import redis.clients.jedis.JedisPool;
+
+@Slf4j
+@Data
+class JedisCacheManager implements CacheManager
+{
+    @NonNull
+    private final JedisCachingProvider cachingProvider;
+
+    @NonNull
+    private final URI uri;
+
+    @NonNull
+    private final ClassLoader classLoader;
+
+    @NonNull
+    private final Properties properties;
+
+    @NonNull
+    private final JedisPool jedisPool;
+
+    private final Map<String, Cache<?, ?>> caches = new HashMap<>();
+
+    private final AtomicBoolean closed = new AtomicBoolean(false);
+
+    @Override
+    public boolean isClosed()
+    {
+        return closed.get();
+    }
+
+    @Override
+    public URI getURI()
+    {
+        return uri;
+    }
+
+    @Override
+    public <K, V, C extends Configuration<K, V>> Cache<K, V> createCache(
+        @NonNull String cacheName, @NonNull C configuration) throws IllegalArgumentException
+    {
+        verifyOpen();
```

Comment:
Hmm, how important is it that after the manager is closed, `createCache()` calls are rejected? As the `verifyOpen()` call is not inside the `synchronized` block, we can't guarantee that the manager is closed by another thread before we create the new cache.

A similar issue exists in the cache itself, now that I think about it. What now?